src/com/base/game/character/GameCharacter.java

```diff
@@ -2918,14 +2889,10 @@ public boolean isFeminine() {
 					isFeminine = true;
 					break;
 				case CLOTHING_FEMININE:
-					isFeminine = true;
-					if(getClothingAverageFemininity()<50)
-						isFeminine = false;
+					isFeminine = getClothingAverageFemininity() >= 50;
```

Comment:
Ehrm.. This seems wrong I presume?

Reply:
Thanks for your interest, but I'm not sure what you mean? I fairly certain those two do the same thing. If the femininity is 50 or above, it's true; if below, it's false.

Also, this changeset is 2+ years old; the code as seen here may have changed substantially since this was merged.

Reply:
No it didn't. I came as far back as here exactly because blame brought me. 

And the part that I'm not understanding is why masculine should turn out positive when femineity is higher.

Reply:
0 is full masculine, 50 is the midpoint, and 100 is full feminine. All values are positive.

Reply:
I see, thanks for replying. 
But then why is there that "greater than" sign for CLOTHING_MASCULINE?

Reply:
`isFeminine = true;
					if(getClothingAverageFemininity()<50)
						isFeminine = false;
					isFeminine = getClothingAverageFemininity() >= 50;`
could be simplified to just this. the other bits are excessive and a waste of computational load, negligible but still a waste.
`isFeminine = getClothingAverageFemininity() >= 50;`

Reply:
mirh: I think you might be misinterpreting what CLOTHING_MASCULINE means. For convenience, I've copied the whole function here:
```
	public boolean isFeminine() {
		boolean isFeminine = body.isFeminine();
		
		if(Femininity.valueOf(getFemininityValue()) == Femininity.ANDROGYNOUS) {
			switch(Main.getProperties().androgynousIdentification){
				case FEMININE:
					isFeminine = true;
					break;
				case CLOTHING_FEMININE:
					isFeminine = getClothingAverageFemininity() >= 50;
					break;
				case CLOTHING_MASCULINE:
					isFeminine = getClothingAverageFemininity() > 50;
					break;
				case MASCULINE:
					isFeminine = false;
					break;
				default:
					break;
			}
		}
		return isFeminine;
	}
```
The values FEMININE, CLOTHING_FEMININE, CLOTHING_MASCULINE, and MASCULINE are determined by the gender settings in the game options. Specifically, this function determines whether an androgynous character should be described as 'feminine' and use feminine words like 'daughter'.

The two options that start with CLOTHING_ look at the femininity/masculinity of that character's clothes to make the decision. In both cases 0 to 49 will be masculine, and 51 to 100 will be feminine. The interesting part is if the value is exactly 50 -- their clothing is completely neutral (or they're not wearing any). In the case of CLOTHING_FEMININE, the value of 50 >= 50, is true, so the character is treated as feminine. In the case of CLOTHING_MASCULINE, the value of 50 > 50 is false, so the character is treated as masculine.

Hopefully that clears things up.

DrZed, I might be miunderstanding what you're trying to say, but I think you've misinterpreted what the show code snippet means. The red highlight shows what this code looked like before, and the green shows after. So the final result is just the green line, as seen in the function as its copied above.

Reply:
Yeah I was tired, and saw a post on something quite old and was slightly irritated due to various factors of the day. Tired due to 37 hours awake at that time, I've slept since then

Reply:
Yeah I was tired, and saw a post on something quite old and was slightly irritated due to various factors of the day. Tired due to 37 hours awake at that time, I've slept since then

Reply:
Holy macaroni, now I see that. Sorry for the misunderstanding.